litr-filters/src/main/java/com/linkedin/android/litr/filter/video/gl/FreeTransformFrameRenderFilter.java

```diff
@@ -0,0 +1,171 @@
+package com.linkedin.android.litr.filter.video.gl;
+
+import android.graphics.PointF;
+import android.opengl.GLES11Ext;
+import android.opengl.GLES20;
+import android.opengl.Matrix;
+
+import androidx.annotation.NonNull;
+
+import com.linkedin.android.litr.filter.GlFrameRenderFilter;
+import com.linkedin.android.litr.render.GlRenderUtils;
+
+/**
+ * Implementation of {@link GlFrameRenderFilter} which can transform (scale, rotate, translate)
+ * source video frame when rendering it onto targetVideoFrame
+ */
+public class FreeTransformFrameRenderFilter implements GlFrameRenderFilter {
```

Comment:
A lot of repeating code from ScaleToFitGiFrameRenderFilter.java, make reviewing hard to focus on the key changes. Plan? I wait to review this file later.

Reply:
I am planning to refactor all the code into common util methods later. I duplicated it for now to isolate new logic and make things easier to review. There are also a lot of similarities with bitmap overlay code, especially for MVP matrix computation.